Data/Primitive/Addr.hs

```diff
@@ -100,3 +107,11 @@ setAddr :: (Prim a, PrimMonad m) => Addr -> Int -> a -> m ()
 {-# INLINE setAddr #-}
 setAddr (Addr addr#) (I# n#) x = primitive_ (setOffAddr# addr# 0# n# x)
 
+-- | Convert an 'Addr' to an 'Int'.
+addrToInt :: Addr -> Int
+{-# INLINE addrToInt #-}
+addrToInt (Addr addr#) = I# (addr2Int# addr#)
```

Comment:
Actually, never mind; I think given that GHC exposes `addr2Int#` and not `addr2Word#` I'd rather follow the prior art.